test/router.js

```diff
@@ -124,6 +124,92 @@ describe('Router', function () {
       .expect(404, cb)
     })
 
+    it('should support promise as route handler', function (done) {
```

Comment:
What does this test? I ran this test without any of your changes and it passed. This test doesn't seem to actually depend on any change, or maybe I'm just missing something.